service/history/shard/context.go

```diff
@@ -624,67 +619,55 @@ func (s *contextImpl) CreateWorkflowExecution(
 		return nil, err
 	}
 
-Create_Loop:
-	for attempt := 0; attempt < conditionalRetryCount && ctx.Err() == nil; attempt++ {
-		if s.isClosed() {
-			return nil, ErrShardClosed
-		}
-		currentRangeID := s.getRangeID()
-		request.RangeID = currentRangeID
-
-		response, err := s.executionManager.CreateWorkflowExecution(ctx, request)
-		switch err.(type) {
-		case nil:
-			// Update MaxReadLevel if write to DB succeeds
-			s.updateMaxReadLevelLocked(transferMaxReadLevel)
-			return response, nil
-		case *types.WorkflowExecutionAlreadyStartedError,
-			*persistence.WorkflowExecutionAlreadyStartedError,
-			*persistence.CurrentWorkflowConditionFailedError,
-			*types.ServiceBusyError:
-			// No special handling required for these errors
-			// We know write to DB fails if these errors are returned
+	if s.isClosed() {
+		return nil, ErrShardClosed
+	}
+	currentRangeID := s.getRangeID()
+	request.RangeID = currentRangeID
+
+	response, err := s.executionManager.CreateWorkflowExecution(ctx, request)
+	switch err.(type) {
+	case nil:
+		// Update MaxReadLevel if write to DB succeeds
+		s.updateMaxReadLevelLocked(transferMaxReadLevel)
+		return response, nil
+	case *types.WorkflowExecutionAlreadyStartedError,
+		*persistence.WorkflowExecutionAlreadyStartedError,
+		*persistence.CurrentWorkflowConditionFailedError,
+		*types.ServiceBusyError:
+		// No special handling required for these errors
+		// We know write to DB fails if these errors are returned
+		return nil, err
+	case *persistence.ShardOwnershipLostError:
+		{
+			// Shard is stolen, trigger shutdown of history engine
+			s.logger.Warn(
+				"Closing shard: CreateWorkflowExecution failed due to stolen shard.",
+				tag.Error(err),
+			)
+			s.closeShard()
 			return nil, err
-		case *persistence.ShardOwnershipLostError:
-			{
-				// RangeID might have been renewed by the same host while this update was in flight
-				// Retry the operation if we still have the shard ownership
-				if currentRangeID != s.getRangeID() {
-					continue Create_Loop
-				} else {
-					// Shard is stolen, trigger shutdown of history engine
-					s.logger.Warn(
-						"Closing shard: CreateWorkflowExecution failed due to stolen shard.",
-						tag.Error(err),
-					)
-					s.closeShard()
-					return nil, err
-				}
-			}
-		default:
-			{
-				// We have no idea if the write failed or will eventually make it to
-				// persistence. Increment RangeID to guarantee that subsequent reads
-				// will either see that write, or know for certain that it failed.
-				// This allows the callers to reliably check the outcome by performing
-				// a read.
-				err1 := s.renewRangeLocked(false)
```

Comment:
this updates range ID.  seems to imply it was used?

Reply:
this is kept and the code is below

Reply:
oooh, I see.  you meant that the range isn't updated in this line:
```go
response, err := s.executionManager.CreateWorkflowExecution(ctx, request)
```
so it can't be different than when it was loaded immediately before that call / it's now immediately after.

yeah, that makes sense.  I want to dig in a bit and verify that, but that does seem very likely true.
any idea if there were other code changes that led to this no longer being reachable?  or has it always been dubious?

Reply:
I did the same change to UpdateWorkflowExeuction and ConflictResolveWorkflowExecution